solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -787,7 +790,8 @@ private void loadInternal() {
     solrClientProvider =
         new HttpSolrClientProvider(cfg.getUpdateShardHandlerConfig(), solrMetricsContext);
     updateShardHandler.initializeMetrics(solrMetricsContext, Attributes.empty());
-    solrClientCache = new SolrClientCache(solrClientProvider.getSolrClient());
+    // We don't pre-initialize SolrClientCache here anymore
+    // It will be created on-demand in getSolrClientCache() when first needed
```

Comment:
Comments like this are are PR review like comments but I don't think should be committed to the code.